packages/contracts-bedrock/README.md

```diff
@@ -1,58 +1,232 @@
+        UserTransaction(User Transaction)
+    end
+
+    subgraph "External Contracts"
+        ExternalERC20(External ERC20 Contracts)
+        ExternalERC721(External ERC721 Contracts)
+    end
+
+    subgraph "Remaining L2 Universe"
+        OtherContracts(Any Contracts and Addresses)
+    end
+
+    L2Node -.->|derives chain from| L1SmartContracts
+    L2Node -->|updates| L1Block
+    L2Node -->|distributes fees to| L1FeeVault
+    L2Node -->|distributes fees to| BaseFeeVault
+    L2Node -->|distributes fees to| SequencerFeeVault
+    L2Node -->|derives from deposits| DepositTransaction
+    L2Node -->|derives from chain data| UserTransaction
+
+    UserTransaction -->|can trigger| OtherContracts
+    DepositTransaction -->|maybe triggers| L2CrossDomainMessenger
+    DepositTransaction -->|can trigger| OtherContracts
+
+    ExternalERC20 <-->|mint/burn/transfer| L2StandardBridge
+    ExternalERC721 <-->|mint/burn/transfer| L2ERC721Bridge
+
+    L2StandardBridge <-->|sends/receives messages| L2CrossDomainMessenger
+    L2ERC721Bridge <-->|sends/receives messages| L2CrossDomainMessenger
+    GasPriceOracle -.->|queries| L1Block
+    L2CrossDomainMessenger -->|sends messages| L2ToL1MessagePasser
+
+    classDef extContracts stroke:#ff9,stroke-width:2px;
+    classDef l2Contracts stroke:#bbf,stroke-width:2px;
+    classDef transactions stroke:#fba,stroke-width:2px;
+    classDef l2Node stroke:#f9a,stroke-width:2px;
+
+    class ExternalERC20,ExternalERC721 extContracts;
+    class L2CrossDomainMessenger,L2ToL1MessagePasser,L2StandardBridge,L2ERC721Bridge l2Contracts;
+    class L1Block,L1FeeVault,BaseFeeVault,SequencerFeeVault,GasPriceOracle l2Contracts;
+    class UserTransaction,DepositTransaction transactions;
+    class L2Node l2Node;
+```
+
+#### Notes for Core L2 Smart Contracts
+
+- Contracts highlighted as "L2 System Contracts" are updated or mutated automatically as part of the chain derivation process. Users typically do not mutate these contracts directly, except in the case of the `FeeVault` contracts where any user may trigger a withdrawal of collected fees to the pre-determined withdrawal address.
+- Smart contracts that sit behind `Proxy` contracts are **highlighted in BLUE**. Refer to the [Smart Contract Proxies](#smart-contract-proxies) section below to understand how these proxies are designed.
+- User interactions for the "L2 Bridge Contracts" have been omitted from this diagram but largely follow the same user interactions described in the architecture diagram for the [Core L1 Smart Contracts](#core-l1-smart-contracts).
+
+### Smart Contract Proxies
 
-[![codecov](https://codecov.io/gh/ethereum-optimism/optimism/branch/develop/graph/badge.svg?token=0VTG7PG7YR&flag=contracts-bedrock-tests)](https://codecov.io/gh/ethereum-optimism/optimism)
+Most L1 and L2 smart contracts for OP Stack chains today sit behind `Proxy` contracts that themselves are managed by a `ProxyAdmin` contract.
```

Comment:
It would be useful if this section also mentioned that L1CrossDomainMessengerProxy is behind a ResolvedDelegateProxy and L1StandardBridgeProxy is behind an L1ChugSplashProxy

Reply:
Will add